src/mame/drivers/dfs500.cpp

```diff
@@ -0,0 +1,1036 @@
+
+    The DSP circuitry was not yet emulated, and the minimal video code in this driver currently
+    simply bypasses the input #1 into the output. Once the serial communication between control panel and
+    maincpu is fixed, we should at least see the video output changing when pressing the BACKGROUND input
+    selector buttons VTR A, VTR B, 3 and 4 (to select ammong the 4 video inputs).
+
+****************************************************************************/
+#define VIDEO_WIDTH 768
+#define VIDEO_HEIGHT 256
+
+#include "emu.h"
+#include "cpu/nec/nec.h"
+#include "machine/gen_latch.h"
+#include "machine/pit8253.h"
+#include "machine/pic8259.h"
+#include "machine/i8251.h"
+#include "machine/upd7004.h"
+#include "sound/beep.h"
+#include "speaker.h"
+#include "imagedev/picture.h"
+#include "screen.h"
+#include "dfs500.lh"
+
+
+class dfs500_state : public driver_device
+{
+public:
+	dfs500_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_cpanelcpu(*this, "cpanelcpu")
+		, m_maincpu(*this, "maincpu")
+		, m_effectcpu(*this, "effectcpu")
+		, m_pit(*this, "pit")
+		, m_pic(*this, "pic")
+		, m_serial1(*this, "serial1")
+		, m_serial2(*this, "serial2")
+		, m_cpanel_serial(*this, "cpanel_serial")
+		, m_cpanel_pit(*this, "cpanel_pit")
+		, m_adc(*this, "adc")
+		, m_rombank1(*this, "rombank1")
+		, m_rombank2(*this, "rombank2")
+		, m_buzzer(*this, "buzzer")
+		, m_screen(*this, "screen")
+		, m_input1(*this, "input1")
+		, m_input2(*this, "input2")
+		, m_input3(*this, "input3")
+		, m_input4(*this, "input4")
+	{
+	}
+
+	void dfs500(machine_config &config);
+
+private:
+	virtual void machine_start() override;
+
+	void cpanelcpu_mem_map(address_map &map);
+	void cpanelcpu_io_map(address_map &map);
+	void maincpu_mem_map(address_map &map);
+	void maincpu_io_map(address_map &map);
+	void effectcpu_mem_map(address_map &map);
+	uint8_t pit_r(offs_t offset);
+	void pit_w(offs_t offset, uint8_t data);
+	uint8_t cpanel_pit_r(offs_t offset);
+	void cpanel_pit_w(offs_t offset, uint8_t data);
+	uint8_t pic_r(offs_t offset);
+	void pic_w(offs_t offset, uint8_t data);
+	void rombank1_entry_w(offs_t offset, uint8_t data);
+	void rombank2_entry_w(offs_t offset, uint8_t data);
+	void input_select_w(offs_t offset, uint8_t data);
+	uint16_t RA0_r(offs_t offset);
+	uint16_t RB0_r(offs_t offset);
+	void WA0_w(offs_t offset, uint16_t data);
+	void WB0_w(offs_t offset, uint16_t data);
+	uint16_t RA1_r(offs_t offset);
+	uint8_t RB1_r(offs_t offset);
+	uint8_t RB2_r(offs_t offset);
+	uint8_t cpanel_reg0_r(offs_t offset);
+	uint8_t cpanel_reg2_r(offs_t offset);
+	void cpanel_reg0_w(offs_t offset, uint8_t data);
+	void cpanel_reg1_w(offs_t offset, uint8_t data);
+	void cpanel_reg2_w(offs_t offset, uint8_t data);
+	uint8_t cpanel_buzzer_r(offs_t offset);
+	u32 screen_update(screen_device &screen, bitmap_rgb32 &bitmap, rectangle const &cliprect);
```

Comment:
Please do not mix shorthand (u8/u16/u32) and long-hand (uint8_t/uint16_t/uint32_t) types.

Reply:
which kind is preferred?

Reply:
The rules for this are:
* Use the standard forms (`uint8_t` etc.) in self-contained things that we want to encourage use of outside MAME (e.g. strformat.{h,cpp} and vecstream.{h,cpp} in src/lib/util)
* Use the abbreviated forms (`u8` etc.) in src/emu
* For drivers and devices, choose a form and stick to it consistently throughout the source file (or files if there are more than one, e.g. the header and source for a device)